awx/api/views.py

```diff
@@ -2874,9 +2874,9 @@ def modernize_launch_payload(self, data, obj):
                     if not isinstance(prompted_value, Iterable) or isinstance(prompted_value, basestring):
                         prompted_value = [prompted_value]
 
-                    # If user gave extra_credentials, special case to use exactly
+                    # If user gave legacy credentials, special case to use exactly
                     # the given list without merging with JT credentials
-                    if key == 'extra_credentials' and prompted_value:
+                    if key in ('credential', 'vault_credential', 'extra_credentials') and prompted_value:
```

Comment:
Support for merging the legacy `credential` and `vault_credential` launch values _was_ being accomplished as a side effect by this block:

https://github.com/ansible/awx/pull/973/files#diff-160977429605d785dda624eb08043a88L381